io-engine/src/bdev/nexus/nexus_bdev.rs

```diff
@@ -607,8 +607,12 @@ impl<'n> Nexus<'n> {
         self.traverse_io_channels(
             sender,
             |chan, _sender| -> ChannelTraverseStatus {
-                chan.reconnect_all();
-                ChannelTraverseStatus::Ok
+                loop {
+                    match chan.reconnect_all() {
```

Comment:
Whilst we're retrying here we're not yielding back to spdk, so how will the async connection connect in the background?

Reply:
hmm, you're right. I'm able to reproduce this with a test intermittently and when it reproduces, I see the reconnect loop running forever. I guess I haven't understood poller behaviour completely yet. I'll see how to yield here and continue spdk iteration from same point.